web/src/components/dashboards/usePerses.ts

```diff
@@ -0,0 +1,99 @@
+import { PersesDashboardMetadata } from './perses-client';
+import { useCallback, useReducer, useRef } from 'react';
+import { fetchPersesDashboardsMetadata } from './perses-client';
+
+const isAbortError = (error: unknown): boolean =>
+  error instanceof Error && error.name === 'AbortError';
+
+type State = {
+  dashboardsData: PersesDashboardMetadata[];
+  isLoadingDashboardsData: boolean;
+  dashboardsError: unknown;
+};
+
+type Action =
+  | {
+      type: 'dashboardsMetadataResponse';
+      payload: { dashboardsData: PersesDashboardMetadata[] };
+    }
+  | {
+      type: 'dashboardsRequest';
+    }
+  | {
+      type: 'dashboardsError';
+      payload: { error: unknown };
+    };
+
+const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case 'dashboardsMetadataResponse':
+      return {
+        ...state,
+        isLoadingDashboardsData: false,
+        dashboardsData: action.payload.dashboardsData,
+      };
+    case 'dashboardsError':
+      return {
+        ...state,
+        isLoadingDashboardsData: false,
+        dashboardsError: action.payload.error,
+      };
+    case 'dashboardsRequest':
+      return {
+        ...state,
+        isLoadingDashboardsData: true,
+        dashboardsData: undefined,
+        dashboardsError: undefined,
+      };
+    default:
+      return state;
+  }
+};
+
+export const usePerses = () => {
+  const dashboardsAbort = useRef<() => void | undefined>();
+
+  const initialState = {
+    dashboardsData: undefined,
+    isLoadingDashboardsData: false,
+    dashboardsError: undefined,
+  };
+
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const { dashboardsData, isLoadingDashboardsData, dashboardsError } = state;
+
+  /**
+   * Lists perses dashboards meta data only (no dashboards specs)
+   */
+  const getPersesDashboards = useCallback(async () => {
```

Comment:
I think instead of having a function to manually retrigger the fetch, it might be better to instead just do polling like how the alerts and silences have

Reply:
This might be a separate JIRA ticket to enhance polling. I create the issue here: [OU-615](https://issues.redhat.com/browse/OU-615).